lib/src/iam/jwks.rs

```diff
@@ -0,0 +1,686 @@
+use crate::err::Error;
+use crate::kvs::Datastore;
+use crate::opt::capabilities::NetTarget;
+use chrono::{DateTime, Duration, Utc};
+use jsonwebtoken::jwk::{Jwk, JwkSet, KeyOperations, PublicKeyUse};
+use jsonwebtoken::{DecodingKey, Validation};
+use once_cell::sync::Lazy;
+use reqwest::{Client, Url};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use std::str::FromStr;
+
+#[cfg(test)]
+static CACHE_EXPIRATION: Lazy<chrono::Duration> = Lazy::new(|| Duration::seconds(1));
+#[cfg(not(test))]
+static CACHE_EXPIRATION: Lazy<chrono::Duration> =
+	Lazy::new(|| match std::env::var("SURREAL_JWKS_CACHE_EXPIRATION") {
```

Comment:
maybe we want to add the units to the env var? so we don't need to go to the docs to know what value is accepted

```suggestion
	Lazy::new(|| match std::env::var("SURREAL_JWKS_CACHE_EXPIRATION_SECS") {
```

Reply:
Good point! I have added units in [bac019c](https://github.com/surrealdb/surrealdb/pull/3198/commits/bac019c484cd91248ba711d6cf31925bdfd69541).